### PR TITLE
tutorial eqnn force field

### PR DESCRIPTION
**Pull request by orielkiss**
------------------------------------------------------------------------------------------------------------

**Title:**
Symmetry-invariant quantum machine learning force fields
**Summary:**
Tutorial on equivariant QML for learning energy and forces in a H2O molecule
**Relevant references:**
https://arxiv.org/abs/2311.11362
**Possible Drawbacks:**
long to train ~10 min

**Related GitHub Issues:**

----
If you are writing a demonstration, please answer these questions to facilitate the marketing process.

* GOALS — Why are we working on this now?
implementation of a recent paper

* AUDIENCE — Who is this for?
Chemistry and QML researchers

* KEYWORDS — What words should be included in the marketing post?
Geometric quantum machine learning, force field

* Which of the following types of documentation is most similar to your file? 
Demos

### CODE REVIEW

File: demonstrations/tutorial_eqnn_force_field.metadata.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            "type": "article",
+            "title": "Exploiting Symmetry in Variational Quantum Machine Learning",
+            "authors": "Johannes Jakob Meyer, Marian Mularski, Elies Gil-Fuster, Antonio Anna Mele, Francesco Arzani, Alissa Wilms, and Jens Eisert",
+            "year": "2023",
+            "publisher": "APS",
+            "journal": "Phys. Rev. X Quantum",
+            "url": "https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.4.010328"
+        }
+    ],
+    "basedOnPapers": [{
```
Comment (KetpuntoG) [inline demonstrations/tutorial_eqnn_force_field.metadata.json:79]: Thanks for opening the PR! Here you should only specify a link to the paper

Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.metadata.json]: thanks, just updated

**Comment by KetpuntoG** [comment]
Good to see it is deploying! 😄
You can check here the demo PR on the website:
https://qml-build-previews.pennylane.ai/pull_request_build_preview/1015/qml/demos/tutorial_eqnn_force_field/

**Comment by orielkiss** [comment]
I think we can get to the reviews. There is a minor problem with the second image I don't know how to fix. But apart from that, it looks in good shape. Thank you! :)

**Comment by KetpuntoG** [comment]
> I think we can get to the reviews. There is a minor problem with the second image I don't know how to fix. But apart from that, it looks in good shape. Thank you! :)

Great, Thanks Oriel 💪
@alvaro-at-xanadu will take a first look at it :)

File: demonstrations/tutorial_eqnn_force_field.py [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+use geometric quantum machine learning to drive molecular dynamics as introduced in recent research
+`[Le 23] <https://arxiv.org/abs/2311.11362>`__.
+
+Introduction
+-----------------------------------
+
+First, let’s talk about **the overall playground of this work: molecular dynamics (MD)**. MD is an
+essential computational simulation method to analyze the dynamics of atoms or molecules in a
+chemical system. The simulations can be used to obtain macroscopic thermodynamic properties of
+ergodic systems. Within the simulation, the Newton equations are numerically integrated. Therefore,
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:18]:
```suggestion
ergodic systems. Within the simulation, the Newton's equations of motion are numerically integrated. Therefore,
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+it is crucial to have access to the forces acting on the constituents of the system or equivalently
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:19]:
```suggestion
it is crucial to have access to the forces acting on the constituents of the system or, equivalently,
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+the potential energy surface, from which we can obtain the atomic forces. Previous research by
+`[Kiss22] <https://iopscience.iop.org/article/10.1088/2632-2153/ac7d3c/meta>`__ have presented variational
+quantum learning models (VQLMs) that were able to learn the potential energy and atomic forces of
+exemplary molecules from *ab initio* reference data.
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:23]:
```suggestion
`[Kiss22] <https://iopscience.iop.org/article/10.1088/2632-2153/ac7d3c/meta>`__ presented variational
quantum learning models (VQLMs) that were able to learn the potential energy and atomic forces of
exemplary molecules from *ab initio* reference data.
```

Comment (alvaro-at-xanadu) [reply demonstrations/tutorial_eqnn_force_field.py:23]: Or "has presented".

File: demonstrations/tutorial_eqnn_force_field.py [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+epsilon = None  # We disable SB for this specific example
+epsilon = jax.lax.stop_gradient(epsilon)  # Uncomment if we wish to train the SB weights as well. 
+
+
+opt_init, opt_update, get_params = optimizers.adam(1e-2)
+net_params = {"params": {"weights": weights, "alphas": alphas, "epsilon": epsilon}}
+opt_state = opt_init(net_params)
+running_loss = []
+#################################
+# We train using stochastic gradient descent
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:456]:
```suggestion
# We train our VQLM using stochastic gradient descent.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+    batch = np.random.choice(np.arange(np.shape(data_train)[0]), batch_size, replace=False)
+
+    loss_data = data_train[batch, ...], E_train[batch, ...], F_train[batch, ...]
+    loss_data_test = data_test, E_test, F_test
+
+    loss, opt_state = train_step(num_batches, opt_state, loss_data)
+
+    E_pred, test_loss = inference(loss_data_test, opt_state)
+    running_loss.append([float(loss), float(test_loss)])
+
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:474]: Code could use some commenting so readers can understand it better.

File: demonstrations/tutorial_eqnn_force_field.py [hunk 7]
```diff
@@ -0,0 +1,602 @@
...
+plt.figure(figsize=(4,4))
+plt.plot(history_loss[:, 0], "r-", label="training error")
+plt.plot(history_loss[:, 1], "b-", label="testing error")
+
+plt.yscale("log")
+plt.xlabel("Optimization Steps", fontsize=fontsize)
+plt.ylabel("Mean Squared Error", fontsize=fontsize)
+plt.legend(fontsize=fontsize)
+plt.show()
+
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:488]: Y label not showing in plot

File: demonstrations/tutorial_eqnn_force_field.py [hunk 8]
```diff
@@ -0,0 +1,602 @@
...
+x = np.arange(len(energy))
+axs[1].plot(x[indices_test], E_pred, "ro")
+axs[1].plot(x[indices_test], energy[indices_test], "k.")
+axs[1].set_xlabel("Data point", fontsize=fontsize)
+
+
+axs[0].legend(fontsize=fontsize)
+plt.show()
+
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:513]: Let's improve the narrative here. What do we see in the plots? What are we plotting?

Comment (alvaro-at-xanadu) [reply demonstrations/tutorial_eqnn_force_field.py:513]: I still feel there's no need for the graph on the right, but feel free to leave it if you think it's useful.

File: demonstrations/tutorial_eqnn_force_field.py [hunk 9]
```diff
@@ -0,0 +1,602 @@
...
+
+######################################################################
+# Force predictions
+# ~~~~~~~~~~~~~~~~~
+#
+# As stated in the beginning, we are interested by obtaining the forces to drive MD simulations. Since
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:519]:
```suggestion
# As stated at the beginning, we are interested by obtaining the forces to drive MD simulations. Since
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 10]
```diff
@@ -0,0 +1,602 @@
...
+# we have access to the potential energy surface, the forces are directly available by taking the
+# gradient
+#
+# .. math:: F_{i,j} = -\nabla_{X_{ij}} E(X, \Theta),
+#
+# where :math:`X_{ij}` contains the :math:`j` coordinate of the :math:`i`-th atom, and :math:`\Theta`
+# are the trainable parameters. In our framework, we can simply do the following.
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:526]: Improve code narrative here, since it's hard to parse. We suddenly have Jacobians where only part is kept and it's not that clear what's going on.

File: demonstrations/tutorial_eqnn_force_field.py [hunk 11]
```diff
@@ -0,0 +1,602 @@
...
+
+######################################################################
+# Conclusions
+# -----------
+#
+# In this demo, we saw how to implement a symmetry-invariant VQLM to learn the energy and forces of
+# small chemical systems and trained it for the specific example of water. The strong points with
+# respect symmetry-agnostic techniques are better generalization, more accurate force predictions,
+# resilience to small data corruption, and reduction in classical pre- and postprocessing, as
+# supported by the original paper `[Le 23] <https://arxiv.org/abs/2311.11362>`__.
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:565]:
```suggestion
# In this demo, we saw how to implement a symmetry-invariant VQLM to learn the energy and forces of
# small chemical systems and trained it for the specific example of water. The strong points with
# respect to symmetry-agnostic techniques are better generalization, more accurate force predictions,
# resilience to small data corruption, and reduction in classical pre- and postprocessing, as
# supported by the original paper `[Le 23] <https://arxiv.org/abs/2311.11362>`__.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 12]
```diff
@@ -0,0 +1,602 @@
...
+#
+# Further work could be devoted to study larger systems by adopting a more systematic fragmentation as
+# discussed in the original paper. As an alternative to building symmetry-invariant quantum
+# architectures, the symmetries could instead be incorporated into the training routine, such as
+# recently proposed by `[Wierichs 23] <https://arxiv.org/abs/2312.06752>`__. Finally, symmetry-aware
+# models could be used to design quantum symmetry functions, which in turn could serve as
+# symmetry-invariant descriptors of the chemical systems within classical deep learning architectures,
+# which can be easily operated and trained at scale.
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_eqnn_force_field.py:573]:
```suggestion
# Further work could be devoted to studying larger systems by adopting a more systematic fragmentation as
# discussed in the original paper. As an alternative to building symmetry-invariant quantum
# architectures, the symmetries could instead be incorporated into the training routine, such as
# recently proposed by `[Wierichs 23] <https://arxiv.org/abs/2312.06752>`__. Finally, symmetry-aware
# models could be used to design quantum symmetry functions, which in turn could serve as
# symmetry-invariant descriptors of the chemical systems within classical deep learning architectures,
# which can be easily operated and trained at scale.
```

**Review by alvaro-at-xanadu — DISMISSED** [review]
I think this is a really good first draft 🚀 ! Thank you for this demo, I learned a lot. I'll summarize some of the main points that I think need to be addressed in order to polish this demo.
- I've found a lot of misuse of language. We need to put this through a language checker before publication, or have a native English speaker check it (I am not a native speaker, so I probably didn't find everything).
- The concepts of invariance and equivariance might need to be introduced in a more systematic and clear way. While some readers may be familiar with it, the words were used way before they were properly introduced, and do remain undefined for certain types of mathematical objects. 
- It was not clear when we chose the size of the molecule. We need to mention explicitly when we're working with two vs. three atoms in the theory.
- The distinction between a transformation and its representation is a bit loose. At all times there is a reference to a $g$, which some people will know is the element of a symmetry/coordinate transformation. But we never define what it is. 
- In general, the mathematical notation has to be better explained. Many variables are loosely defined. 
- Towards the end, the code lacks narrative. While some things may  be straightforward, it's good to guide the reader through each block of code and summarize what's going on.
- Some discussion of what we observe in plots would be nice too. 
- Some graphs lack a y-axis label and I have no idea why.
- Citations and bibliography need to be fixed to be in the style of other demos.

I know it's a lot of bullet points 😅 , but they should all be easy to address! Great work 💪🏼 !

**Comment by KetpuntoG** [comment]
Notice that we have also [this demo](https://pennylane.ai/qml/demos/tutorial_equivariant_graph_embedding/). You can link some concepts like equivariant or invariant

File: demonstrations/tutorial_eqnn_force_field.py [hunk 13]
```diff
@@ -0,0 +1,602 @@
...
+transformation acting at the data level. Here, we call the symmetry representation on the data level
+:math:`V_g`. For the cases of a diatomic molecule (e.g. LiH) and a triatomic molecule of two atom
+types (e.g. H2O), panel (a) of the following figure displays the descriptions of the chemical
+systems while the general circuit formulation of the corresponding symmetry-invariant VQLM is shown
+in panel (b), while panel (a) shows the input of the model.
+
+ .. figure:: ../_static/demonstration_assets/eqnn_force_field/siVQLM_monomer.png
+    :align: center
+    :width: 70%
+
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:66]: i am not sure how to resolve that, any idea?

File: demonstrations/tutorial_eqnn_force_field.py [hunk 14]
```diff
@@ -0,0 +1,602 @@
...
+    # exact (bool): flag for exact quantum simulation of the Heisenberg dynamics
+
+    hamiltonian = jnp.einsum("i,ijk", data, sigmas)  # Heisenberg Hamiltonian
+    U = jax.scipy.linalg.expm(-1.0j * alpha * hamiltonian / 2)
+
+    if exact:
+        qml.QubitUnitary(U, wires=wires, id="E")
+    else:  # IL: I guess we can just exclude this case then
+        # Question for reviewer: should we incluse this case (not covered in the paper)
+        # We have to fixe the correct wires ..., also i think it does not work with jax ... TBD
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:181]: works for us

File: demonstrations/tutorial_eqnn_force_field.py [hunk 15]
```diff
@@ -0,0 +1,602 @@
...
+
+
+def trainable_layer(weight, wires, exact=True):
+    hamiltonian = jnp.einsum("ijk->jk", sigmas_sigmas)
+    U = jax.scipy.linalg.expm(-1.0j * weight * hamiltonian)
+
+    if exact:
+        qml.QubitUnitary(U, wires=wires, id="U")
+    else:  # IL: I guess we can exclude this case
+        qml.TrotterProduct(Heisenberg, weight, order=1, n=1)
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:243]: done

File: demonstrations/tutorial_eqnn_force_field.py [hunk 16]
```diff
@@ -0,0 +1,602 @@
...
+# within this demo.
+#
+
+############ Setup ##############
+D = 6  # Depth of the model
+B = 1  # Number of repetion inside a trainable layer
+rep = 2  # Number of repeated vertical encoding
+exact = True
+
+atoms = 2  # Number of atoms, here two since we fixe the oxygen at the origin
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:279]: yes we have 3 atoms but only work with 2 active atoms since we fix the oxygen at the center. we will remove LiH entirely at it is confusing and make it more clear we are dealing with H2O only

File: demonstrations/tutorial_eqnn_force_field.py [hunk 17]
```diff
@@ -0,0 +1,602 @@
...
+num_qubits = atoms * rep
+#################################
+
+
+dev = qml.device("default.qubit.jax", wires=num_qubits)
+
+
+@qml.qnode(dev, interface="jax")
+def eqnn(data, params):
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:288]: done

File: demonstrations/tutorial_eqnn_force_field.py [hunk 18]
```diff
@@ -0,0 +1,602 @@
...
+    data, E_target, F_target = loss_data
+    net_params = get_params(opt_state)
+
+    E_pred = v_eqnn(data, net_params)
+    l = mse_loss(E_pred, E_target)
+
+    return E_pred, l
+
+#################################
+# Parameters initialization
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:433]: adding warm start and no SB

**Comment by KetpuntoG** [comment]
Thanks @orielkiss ! Catalina will do the second review 😊

**Comment by KetpuntoG** [comment]
Just to say that we did not forget to take a look to this demo 💪 With QHack we have a little less capacity but we will get to it soon!

File: demonstrations/tutorial_eqnn_force_field.py [hunk 19]
```diff
@@ -0,0 +1,627 @@
...
+            )
+
+    return qml.expval(Observable)
+
+
+######################################################################
+# Simulation for the water molecule
+# -----------------------------------
+#
+# We start by dowloading the `dataset <https://zenodo.org/records/2634098>`__, which we have prepared for
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:349]:
```suggestion
# We start by downloading the `dataset <https://zenodo.org/records/2634098>`__, which we have prepared for
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 20]
```diff
@@ -0,0 +1,627 @@
...
+# convenience as a python ndarray. In the following, we will load, preprocess and split the data into a
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:350]:
```suggestion
# convenience as a Python ndarray. In the following, we will load, preprocess and split the data into a
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 21]
```diff
@@ -0,0 +1,627 @@
...
+
+E_train, E_test = (energy[indices_train, 0], energy[indices_test, 0])
+F_train, F_test = forces[indices_train, ...], forces[indices_test, ...]
+data_train, data_test = (
+    jnp.array(positions[indices_train, ...]),
+    jnp.array(positions[indices_test, ...]),
+)
+
+#################################
+# We will know define the cost function and how to train the model using jax. We will use the mean-square-error loss function.
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:392]:
```suggestion
# We will know define the cost function and how to train the model using Jax. We will use the mean-square-error loss function.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 22]
```diff
@@ -0,0 +1,627 @@
...
+# To speed up the computation, we use the decorator ``jax.jit`` to compile this execution just-in-time. This means the first execution will typically take a little longer with the
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:393]:
```suggestion
# To speed up the computation, we use the decorator ``@jax.jit`` to do just-in-time compilation for this execution. This means the first execution will typically take a little longer with the
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 23]
```diff
@@ -0,0 +1,627 @@
...
+# benefit that all following executions will be significantly faster, see the `jax docs on jitting <https://jax.readthedocs.io/en/latest/jax-101/02-jitting.html>`_.
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:394]:
```suggestion
# benefit that all following executions will be significantly faster, see the `Jax docs on jitting <https://jax.readthedocs.io/en/latest/jax-101/02-jitting.html>`_.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 24]
```diff
@@ -0,0 +1,627 @@
...
+    data, E_target, F_target = loss_data
+    net_params = get_params(opt_state)
+
+    E_pred = vec_vqlm(data, net_params)
+    l = mse_loss(E_pred, E_target)
+
+    return E_pred, l
+
+#################################
+# **Parameters initialization:**
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:442]:
```suggestion
# **Parameter initialization:**
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 25]
```diff
@@ -0,0 +1,627 @@
...
+#
+# We initiliase the model at the identity by setting the initial parameters to 0, except the first one which is chosen uniformly.
+# This ensures that the circuit is shallow at the beginning and has less chance of suffering from the barren plateau phenomenon. Moreover,
+# we disable the symmetry breaking strategy, as it is mainly useful for larger systems.
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:446]:
```suggestion
# we disable the symmetry-breaking strategy, as it is mainly useful for larger systems.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 26]
```diff
@@ -0,0 +1,627 @@
...
+plt.legend(fontsize=fontsize)
+plt.tight_layout()
+plt.show()
+######################################################################
+# Energy predictions
+# ~~~~~~~~~~~~~~~~~~
+#
+# We first inspect the quality of the energy predictions. The exact test energy points are shown in black, while the predictions are in red.
+# On the left, we see the exact data against the oredicted ones (so the red points should be in the diagonal line), while the right plots show the
+# energy as a scatter plot. The model is able to make fair predictions, especially near the equilibrium possition. However, a few points in the higher energy range
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:512]:
```suggestion
# energy as a scatter plot. The model is able to make fair predictions, especially near the equilibrium position. However, a few points in the higher energy range
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 27]
```diff
@@ -0,0 +1,627 @@
...
+# could be improved, e.g. by using a deeper model.
```
Comment (CatalinaAlbornoz) [inline demonstrations/tutorial_eqnn_force_field.py:513]: Is this true? Deeper models don't necessarily improve performance. Did you make a test that shows this or is it in a paper?

**Review by CatalinaAlbornoz — CHANGES_REQUESTED** [review]
Looks good but requires some changes.

File: demonstrations/tutorial_eqnn_force_field.py [hunk 28]
```diff
@@ -0,0 +1,627 @@
...
+weights = jnp.array(weights)
+
+# Encoding weights
+alphas = jnp.array(np.ones((num_qubits, D + 1)))
+
+# Symmetry-breaking (SB)
+np.random.seed(42)
+epsilon = jnp.array(np.random.normal(0, 0.001, size=(D, num_qubits)))
+epsilon = None  # We disable SB for this specific example
+epsilon = jax.lax.stop_gradient(epsilon)  # Uncomment if we wish to train the SB weights as well.
```
[the remark this answers is not in the document]
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:459]: indeed, i meant comment instead of uncomment

File: demonstrations/tutorial_eqnn_force_field.py (hunk 27 above)
Comment (orielkiss) [reply demonstrations/tutorial_eqnn_force_field.py:513]: in the paper, we go to D=20 and obtain a slightly better performance, at the cost of a longer training time. so we know for sure the performance can be improved in that way. I have added a comment referring to the paper

**Comment by orielkiss** [comment]
Hi! Thanks a lot for your feedback @CatalinaAlbornoz. I have just committed to all the language changes you requested. I have also precised that in the paper better performances are obtained with a deeper model. Also, I have better explained the part with the swapping in line 245 as
"This is fulfilled by coupling only the qubits with are neighbors with respect to the 1-2-3-4-1 ring topology, leading to the following operation".

**Comment by KetpuntoG** [comment]
Thank you so much @orielkiss ! 🚀 I'll talk to the designer to create a Thumbnail and marketing to set a publication date. I estimate it will be within approximately 3-4 weeks. As soon as I have news I'll tell you :)

**Review by CatalinaAlbornoz — APPROVED** [review]

**Comment by KetpuntoG** [comment]
Hey ! for the publication it is possible that marketing may want to include your username in the post. Could you share with me your @username from Twitter (optional)?

**Review by KetpuntoG — APPROVED** [review]
🚀🚀🚀

**Comment by orielkiss** [comment]
Hi @KetpuntoG ! Thank you for the great work, i really feel our demo improved a lot from the first version. Sure you can use my twitter name @oriel_kiss for marketing (and i think Isabel does not have twitter)

**Comment by KetpuntoG** [comment]
Thanks to you for the good work! Tomorrow it will appear on the website but we will officially announce it next week. I will tell you the exact date as soon as marketing confirm me

**Comment by KetpuntoG** [comment]
Hey! Marketing has scheduled the post for March 20 :)

**Comment by orielkiss** [comment]
Awesome! i just remarked something weird happened at training time that never happened with the preview: it seems the training was stopped and then started again with random parameters. is the process different from the preview and the official website? (not a big deal otherwise)

**Comment by larawatson** [comment]
@orielkiss Awesome! Just to follow up on Guillermo's note, we are pushing to Marketing of this by 1 week, so it will be out on Thursday March 28th

**Comment by larawatson** [comment]
@orielkiss Hey Oriel,

We have to push the Marketing on this one slightly. I will get back to you with a new date shortly, thanks for your patience!

**Comment by larawatson** [comment]
Hey @orielkiss this will be marketed on April 2nd :)
